src/lit-html.ts

```diff
@@ -145,21 +159,31 @@ export class Template {
     this.element.innerHTML = this._getHtml(strings, svg);
     // Edge needs all 4 parameters present; IE11 needs 3rd parameter to be null
     const walker = document.createTreeWalker(
-        this.element.content, 5 /* elements & text */, null as any, false);
+        this.element.content,
+        133 /* NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT */,
+        null as any, false);
     let index = -1;
     let partIndex = 0;
-    const nodesToRemove = [];
+    const nodesToRemove: Node[] = [];
+
+    // The actual previous node, accounting for removals: if a node is removed it will
+    // never be the previousNode.
+    let previousNode: Node | undefined;
+    // Used to set previousNode at the top of the loop.
+    let currentNode: Node | undefined;
 
     while (walker.nextNode()) {
       index++;
-      const node = walker.currentNode as Element;
-      if (node.nodeType === 1 /* ELEMENT_NODE */) {
-        if (!node.hasAttributes())
+      previousNode = currentNode;
+      const node = currentNode = walker.currentNode as Element;
+      if (node.nodeType === 1 /* Node.ELEMENT_NODE */) {
+        if (!node.hasAttributes()) {
```

Comment:
nit: you could remove this `hasAttributes()` call as later you just loop on a potentially empty list of `node.attributes`. This should be _faster_ (invoking hasAttributes/getAttribute can be slow)

Reply:
I was told by @sorvell that we had found `hasAttribtues()` to be faster

Reply:
True that! https://jsperf.com/hasattributes-vs-attributes/1